apps/coscrad-frontend/src/store/slices/resources/shared/connected-resources/connected-resources-panel.presenter.tsx

```diff
@@ -1,21 +1,69 @@
+import { ResourceType } from '@coscrad/api-interfaces';
+import { SelectedResourceContainer } from '../../../../../components/higher-order-components/selected-resources.container';
+import { thumbnailResourceDetailPresenterFactory } from '../../../../../components/resources/factories/thumbnail-resource-detail-presenter-factory';
 import { ConnectedResource } from '../../../notes/hooks';
-import { ResourceDetailThumbnail } from './resource-detail-thumbnail';
+
+type SelectedResourcesMap = Map<ResourceType, string[]>;
 
 interface Props {
     data: ConnectedResource[];
 }
 
+/**
+ * Strictly speaking, this is not a presenter. It is a nested container.
+ * We rely on
+ * - first fetching the resource
+ * - then given the resource, fetching its connections
+ * - for each connected resource type, fetching the given resources in the `SelectedResourcesContainer`
+ */
 export const ConnectedResourcesPanelPresenter = ({
     data: connectedResources,
-}: Props): JSX.Element => (
-    <div>
-        <h2>Connected Resources</h2>
-        {connectedResources.map(({ compositeIdentifier, selfContext, otherContext }) => (
-            <div>
-                <ResourceDetailThumbnail {...compositeIdentifier} />
-                self context: {JSON.stringify(selfContext)}
-                other context: {JSON.stringify(otherContext)}
-            </div>
-        ))}
-    </div>
-);
+}: Props): JSX.Element => {
+    /**
+     * We don't necessarily need to show resources of every resource type in
```

Comment:
Just wanted to clarify what you mean here, is this because only the types relevant or connected to the given parent resource will be loaded?

Reply:
Yea, exactly. I have reworded this comment for clarity.